app/src/main/java/de/westnordost/streetcomplete/quests/barrier_opening/AddGateWidthForm.kt

```diff
@@ -0,0 +1,96 @@
+
+import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AlertDialog
+import androidx.core.view.isGone
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.databinding.QuestLengthBinding
+import de.westnordost.streetcomplete.osm.Length
+import de.westnordost.streetcomplete.osm.hasDubiousRoadWidth
+import de.westnordost.streetcomplete.quests.AbstractArMeasureQuestForm
+import de.westnordost.streetcomplete.quests.width.WidthAnswer
+import de.westnordost.streetcomplete.screens.measure.ArSupportChecker
+import de.westnordost.streetcomplete.view.controller.LengthInputViewController
+import org.koin.android.ext.android.inject
+
+class AddGateWidthForm : AbstractArMeasureQuestForm<WidthAnswer>() {
+
+    override val contentLayoutResId = R.layout.quest_length
+    private val binding by contentViewBinding(QuestLengthBinding::bind)
+    private val checkArSupport: ArSupportChecker by inject()
+    private var isARMeasurement: Boolean = false
+    private lateinit var lengthInput: LengthInputViewController
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        savedInstanceState?.let { isARMeasurement = it.getBoolean(AR) }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val isBarrier = element.tags["barrier"] in GATEWAYS
+        val explanation = if (isBarrier) getString(R.string.quest_barrier_opening_width_explanation) else null
+        binding.widthExplanationTextView.isGone = explanation == null
+        binding.widthExplanationTextView.text = explanation
+
+        lengthInput = binding.lengthInput.let {
+            LengthInputViewController(it.unitSelect, it.metersContainer, it.metersInput, it.feetInchesContainer, it.feetInput, it.inchesInput)
+        }
+        lengthInput.unitSelectItemResId = R.layout.spinner_item_centered_large
+        lengthInput.isCompactMode = true
+        lengthInput.maxFeetDigits = if (isBarrier) 3 else 2
+        lengthInput.maxMeterDigits = Pair(if (isBarrier) 2 else 1, 2)
+        lengthInput.selectableUnits = countryInfo.lengthUnits
+        lengthInput.onInputChanged = {
+            isARMeasurement = false
+            checkIsFormComplete()
+        }
+        binding.measureButton.isGone = !checkArSupport()
+        binding.measureButton.setOnClickListener { takeMeasurement() }
+    }
+
+    private fun takeMeasurement() {
+        val lengthUnit = lengthInput.unit ?: return
+        takeMeasurement(lengthUnit, false)
+    }
+
+    override fun onMeasured(length: Length) {
+        lengthInput.length = length
+        isARMeasurement = true
+    }
+
+    override fun onClickOk() {
+        val length = lengthInput.length!!
+        val newTags = element.tags + ("width" to length.toMeters().toString())
+        if (hasDubiousRoadWidth(newTags) != true) {
```

Comment:
hasDubiousRoadWidth seems unfit to check dubious values for gates on paths etc.

Reply:
Should I remove this completely, or amend the values?  I wondered if it would be useful if an update included a measurement in the 100's of metres - missing decimal point.

Reply:
The first time I reviewed the code, I misread. I thought that as long as `hasDubiousRoadWidth` does not return false, it will always ask for confirmation.

But currently, it only displays that confirmation dialog if `hasDubiousRoadWidth(newTags)` is `true`, so that's fine. No change necessary, IMO